### PR TITLE
docs(eslint-plugin):improve docs [consistent-type-exports]

### LINKED ISSUE #4769: [consistent-type-exports] Improve docs
### Suggested Changes

The docs look a little... wrong to me.

> This rule aims to standardize the use of type exports style across a codebase.
>
> Given a class `Button`, and an interface `ButtonProps`, examples of code:

But the actual example is missing.

Further down, the "correct" example for `fixMixedExportsWithInlineTypeSpecifier`:

```ts
export { Button } from 'some-library';
export type { ButtonProps } from 'some-library';
export { Button, type ButtonProps } from 'some-library';
```

Is obviously duplicate export. Is that intended?

Furthermore, it may be beneficial to point out in "When Not To Use It" that `isolatedModules` already covers this:

```
Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.
```

### Affected URL(s)

https://typescript-eslint.io/rules/consistent-type-exports

### PR DESCRIPTION
**Pull request by kmin-jeong**
## PR Checklist

-   [v] Addresses an existing issue: fixes #4769 
-   [v] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [v] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

I remove duplicate exports in the "correct" example for fixMixedExportsWithInlineTypeSpecifier. and I add about 'isolatedModules' to "When Not To Use It".
I wish reviewers check to see if it has been modified to fit the context.

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:80]: Probably a minor rewording so it goes more along the lines of "If you use `--isolatedModules` you don't need this rule since TS would enforce type-export specifiers"

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Missing the example in the "rule details" section

File: packages/eslint-plugin/CHANGELOG.md
Comment (Josh-Cena) [inline packages/eslint-plugin/CHANGELOG.md:8]: Remove this?

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:26]: The example should illustrated re-exporting types. For example:

```suggestion
interface ButtonProps {
  onClick: () => void;
}

class Button implements ButtonProps {
  onClick() {
    console.log('button!');
  }
}

export { Button };
export type { ButtonProps };
```

Probably we should also give an invalid example, like `export { Button, ButtonProps };`

Also, I believe code style can be fixed with Prettier. Please remember to format files before committing.

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:93]: Not critical, but I think it's clearer this way:

```suggestion
- If you use `--isolatedModules`, you don't need this rule since the compiler would error if a type is not re-exported using `export type`.
```

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:29]: Please refer to how other rule docs are written. You should use two tabs, one incorrect and one correct.

Comment (kmin-jeong) [reply packages/eslint-plugin/docs/rules/consistent-type-exports.md]: @Josh-Cena Should I write more explanations about the wrong example?

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:16]: The `<!--tabs-->` comment before the heading is actually useful. It will turn the headings into tabs. See https://typescript-eslint.io/rules/array-type#array for example.

Comment (kmin-jeong) [reply packages/eslint-plugin/docs/rules/consistent-type-exports.md:18]: ok! I added  above examples

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:111]: I imagine some teams would want this rule as a stylistic choice regardless of their `--isolatedModules` setting.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for sending! +1 to what Josh-Cena said, and one little change from me please. 🙂

**Comment by kmin-jeong** [comment]
@JoshuaKGoldberg @Josh-Cena @bradzacher  
Is there anything else I need to modify?

File: packages/eslint-plugin/CHANGELOG.md
Comment (Josh-Cena) [inline packages/eslint-plugin/CHANGELOG.md:8]: Again, I think this needs to be removed

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:111]: Split this into two items

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

**Comment by kmin-jeong** [comment]
@Josh-Cena  Does it still look reverted in changelog.md?

**Comment by Josh-Cena** [comment]
You can check yourself in https://github.com/typescript-eslint/typescript-eslint/pull/4792/files :)

**Comment by kmin-jeong** [comment]
@Josh-Cena  I commit the code with the message "fix: revert change log" yesterday after reverting changelog.md. 
I compared and fixed the changelog.md in the typescript-eslint branch and the changelog in the kminjeong branch
Is there any sentence that needs to be removed or reverted from the changelog?

**Comment by Josh-Cena** [comment]
You shouldn't change changelog.md at all.

**Comment by kmin-jeong** [comment]
@Josh-Cena 
Then, have the required modifications been satisfied?

**Comment by Josh-Cena** [comment]
Yes, this looks good to me. Just waiting for one of the maintainers to review it :)

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/consistent-type-exports.md:112]: @Josh-Cena my intent here was for these to be on the same line. 

```suggestion
- If you use `--isolatedModules` the compiler would error if a type is not re-exported using `export type`. If you also don't wish to enforce one style over the other, you can disable this rule.
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Thanks @kmin-jeong and @Josh-Cena! This looks great! 🙌

File: packages/eslint-plugin/docs/rules/consistent-type-exports.md
Comment (Josh-Cena) [reply packages/eslint-plugin/docs/rules/consistent-type-exports.md]: Really...? They look like completely disjoint points though.

**Pull request merged**
